src/main/java/games/strategy/triplea/oddscalc/OddsCalculatorPanel.java

```diff
@@ -311,7 +311,7 @@ private void updateStats() {
       throw new IllegalStateException("Wrong thread");
     }
     final AtomicReference<AggregateResults> results = new AtomicReference<>();
-    final WaitDialog dialog = new WaitDialog(this, "Calculating Odds", e -> calculator.cancel());
+    final WaitDialog dialog = new WaitDialog(this, "Calculating Odds", () -> calculator.cancel());
```

Comment:
This statement can be simplified to calculator::cancel

Reply:
I actually did that at first.  However, `calculator` is not `final` and is `null` at the time this line is executed.  Thus, using `calculator::cancel` results in an NPE when the lambda is invoked.  `() -> calculator.cancel()` causes `calculator` to be evaluated when the lambda is invoked rather than when the lambda is created, so it picks up the future non-`null` value.

Reply:
Thanks for clearing that up.
Perhaps a comment would be good so no one changes this by accident.

Reply:
Yup, will do.

Reply:
Just so it's clear what's going on here, I created a simple class ([source](https://github.com/triplea-game/triplea/files/1455081/LambdaTest.java.txt)) that demonstrates the difference between `() -> calculator.cancel()` and `calculator::cancel` in this context, and then disassembled it ([source](https://github.com/triplea-game/triplea/files/1455082/LambdaTest.asm.txt)) to get the generated bytecode.

Here's a mixed source-bytecode listing of the code of interest:

```
//  9:     final Runnable r1 = () -> calculator.cancel();
         0: aload_0
         1: invokedynamic #20,  0             // InvokeDynamic #0:run:(Lgames/strategy/triplea/oddscalc/LambdaTest;)Ljava/lang/Runnable;
         6: astore_1
// 10:     r1.run();
         7: aload_1
         8: invokeinterface #21,  1           // InterfaceMethod java/lang/Runnable.run:()V
// 11:     final Runnable r2 = calculator::cancel;
        13: aload_0
        14: getfield      #25                 // Field calculator:Lgames/strategy/triplea/oddsCalculator/ta/ConcurrentOddsCalculator;
        17: invokedynamic #29,  0             // InvokeDynamic #1:run:(Lgames/strategy/triplea/oddsCalculator/ta/ConcurrentOddsCalculator;)Ljava/lang/Runnable;
        22: astore_2
// 12:     r2.run();
        23: aload_2
        24: invokeinterface #21,  1           // InterfaceMethod java/lang/Runnable.run:()V
```

The reference assigned to `r1` is a method that is bound to `this` when this line is executed.  It's only when the body of that method is run is `this.calculator` evaluated.

However, the reference assigned to `r2` is a method that is bound to `this.calculator` when this line is executed.  Thus, if `this.calculator` is `null` at this point, the method reference gets created with a `null` call site, and an NPE will be raised when it is invoked.

Reply:
Comment added as part of rebased ac4b492.